File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.cpp
class Solution {
public:
    int sumOddLengthSubarrays(vector<int>& arr) {
        
        
        int ans=0;
        
        for(int i=0;i<arr.size();i++)
        {
            for(int j=i;j<arr.size();j++)
            {
                for(int k=i;k<=j;k++)
                {
                    int size=(j-i+1);
                    if(size&1)
                    {
                        ans+=arr[k];
                    }
                }
            }
        }
        
        return ans;
        
    }
};

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.java
class Solution {

    public int sumOddLengthSubarrays(int[] arr) {
        int n = arr.length;

        int[] preSum = new int[n];

        preSum[0] = arr[0];

        for (int i = 1; i < n; i++) {
            preSum[i] = preSum[i - 1] + arr[i];
        }
        int sum = 0;
        int i = 0;
        while (i < arr.length) {
            sum += preSum[i];
            for (int j = 1; j < preSum.length - i; j++) {
                sum += preSum[i + j] - preSum[j - 1];
            }
            i += 2;
        }

        return sum;
    }
}
